lib/public/views/Runs/ActiveColumns/runDetectorsAsyncQcActiveColumns.js

```diff
@@ -12,6 +12,62 @@
  */
 import { h } from '/js/src/index.js';
 import { frontLink } from '../../../components/common/navigation/frontLink.js';
+import { formatFloat } from '../../../utilities/formatting/formatFloat.js';
+import { isContrastBlack } from '../../../components/common/badge.js';
+import { ChartColors } from '../../Statistics/chartColors.js';
+
+/**
+ * Render QC summary for given run and detector
+ * @param {number} runNumber run number
+ * @param {number} dplDetectorId id of the detector
+ * @param {QcSummary} qcSummary QC summary
+ * @return {Component} component
+ */
+const runDetectorAsyncQualityDisplay = (runNumber, dplDetectorId, qcSummary) => {
+    if (!qcSummary?.[runNumber]?.[dplDetectorId]) {
+        return h('.btn.white.bg-primary', 'QC');
+    }
+
+    const {
+        badEffectiveRunCoverage = null,
+        explicitlyNotBadEffectiveRunCoverage = null,
+        missingVerificationsCount = 0,
+    } = qcSummary[runNumber][dplDetectorId];
+
+    const notBadPercentageFormat = [
+        formatFloat((1 - badEffectiveRunCoverage) * 100, { precision: 0 }),
+        missingVerificationsCount ? h('sub', '!') : null,
+    ];
+
+    /**
+     * Format QC summary with color
+     * @param {*} content content
+     * @param {string} [configuration.color] background color
+     * @returns {Component} QC summary display
+     */
+    const getQcSummaryDisplayWithColor = (content, { color }) => h(
+        'span.btn.w-100',
+        { style: {
```

Comment:
I believe it would be best to extract these as classes and use from the CSS files so that flags styles can be reused in the future